core/src/model/PlayerWeapon.java

```diff
@@ -0,0 +1,95 @@
+package model;
+
+import com.dongbat.jbump.*;
+
+import java.util.*;
+
+import static java.lang.Math.*;
+
+public abstract class PlayerWeapon {
+
+    private float weaponSpeed;
+    private float weaponDamage;
+    private float weaponRange;
+    private float weaponWidth;
+    private double weaponAngle;
+    private int weaponRotations;
+    private World<IEntity> world;
+    private CollisionFilter filter;
+    private Set<Entity> isKnown = new HashSet<Entity>();
+    private ArrayList<ItemInfo> items = new ArrayList<>();
+    private List<Entity> entities = new LinkedList<>();
```

Comment:
This is only ever used inside of a method and immediately cleared. Is it necessary to store the reference? Maybe move this to be created inside the method and to be garbage collected when no longer needed.

Reply:
i believe this was created to avoid a memory leak